api/v1alpha1/tenantcontrolplane_registrysettings_funcs.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Clastix Labs
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"fmt"
+)
+
+func (r *RegistrySettings) buildContainerImage(name, tag string) (image string) {
+	if len(r.Registry) > 0 {
+		image = fmt.Sprintf("%s/", r.Registry)
+	}
```

Comment:
Since registry defaults to `registry.k8s.io`, what is the rationale behind this check? To make sure the user is not passing an empty string?

Reply:
Being allowed to specify the default docker.io container registry, but now I noticed that we have to use a pointer, otherwise with an empty string the default one will be always used.

Reply:
I guess people can explicitly say `docker.io/nginx:latest` if they feel to use the docker hub.

Reply:
You're right, so we can reduce the toil of additional checks.